dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/RuntimeClientLoader.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dashbuilder.client;
+
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.google.gwt.user.client.Window;
+import org.dashbuilder.client.navigation.NavigationManager;
+import org.dashbuilder.client.perspective.generator.RuntimePerspectiveGenerator;
+import org.dashbuilder.client.plugins.RuntimePerspectivePluginManager;
+import org.dashbuilder.shared.model.RuntimeModel;
+import org.dashbuilder.shared.model.RuntimeServiceResponse;
+import org.dashbuilder.shared.service.RuntimeModelService;
+import org.jboss.errai.common.client.api.Caller;
+import org.uberfire.mvp.Command;
+
+@ApplicationScoped
+public class RuntimeClientLoader {
+
+    public static final String IMPORT_ID_PARAM = "import";
+
+    private Caller<RuntimeModelService> runtimeModelServiceCaller;
+
+    RuntimePerspectiveGenerator perspectiveEditorGenerator;
+
+    RuntimePerspectivePluginManager runtimePerspectivePluginManager;
+
+    NavigationManager navigationManager;
+
+    public RuntimeClientLoader() {
+        // do nothing
+    }
+
+    @Inject
+    public RuntimeClientLoader(Caller<RuntimeModelService> importModelServiceCaller,
+                               RuntimePerspectiveGenerator perspectiveEditorGenerator,
+                               RuntimePerspectivePluginManager runtimePerspectivePluginManager,
+                               NavigationManager navigationManager) {
+        this.runtimeModelServiceCaller = importModelServiceCaller;
+        this.perspectiveEditorGenerator = perspectiveEditorGenerator;
+        this.runtimePerspectivePluginManager = runtimePerspectivePluginManager;
+        this.navigationManager = navigationManager;
+    }
+
+    public void load(Consumer<RuntimeServiceResponse> responseConsumer,
+                     BiConsumer<Object, Throwable> error) {
+        String importID = getImportId();
+        runtimeModelServiceCaller.call((RuntimeServiceResponse response) -> {
+            response.getRuntimeModelOp().ifPresent(this::registerModel);
```

Comment:
If is not present then the caller will decide what to do, in this case it will route to the list dashboards or empty screen. The feedback is indeed the empty optional. But if there's a runtime model we must register it (perspectives and navigation) before responding